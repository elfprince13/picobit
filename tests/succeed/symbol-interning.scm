(define A 'hello)
(define C 'potato)
((lambda (B D)
 (print A)
 (print B)
 (print C)
 (displayln D)
 (displayln "ABCDabcdHello Potato")
 (print (eq? A C))
 (print (eq? A (string->symbol "hello")))
 (print (eq? A (string->symbol "catapult")))
 (print (eq? C (string->symbol "toothpaste")))
 (print (eq? C (string->symbol "potato")))
 (print (eq? B (string->symbol "cheeseburger")))
 (print (eq? D (string->symbol "doorknob")))
 (print (eq? D (string->symbol "equalizer")))
 (print (eq? D (string->symbol "smithy")))
 (print (eq? D (string->symbol "vignette")))) 'cheeseburger 'doorknob)