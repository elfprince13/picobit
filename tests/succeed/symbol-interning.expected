'hello
'cheeseburger
'potato
'doorknob
ABCDabcdHello Potato
#f
#t
#f
#f
#t
#t
#t
#f
#f
#f
